### PR TITLE
choose_max_value requirements

### PR DESCRIPTION
**Pull request by kamalca**
Requirements that are CountSpaces / IntRanges can now specify to select the max value instead of the min value.

For example, test cases like verify_synthetic_provision_with_max_nics can specify that they want the max nics for a given capability instead of hard coding 8 as it currently does.

### CODE REVIEW

File: lisa/microsoft/testsuites/gpu/gpusuite.py [hunk 1]
```diff
@@ -123,11 +126,17 @@ def verify_gpu_provision(self, node: Node, log: Logger) -> None:
...
         # max GPU count available in Azure
-        requirement=simple_requirement(min_gpu_count=8),
+        requirement=node_requirement(
+            node=schema.NodeSpace(gpu_count=IntRange(min=2, choose_max_value=True))
+        ),
         priority=3,
     )
     def verify_max_gpu_provision(self, node: Node, log: Logger) -> None:
-        _gpu_provision_check(8, node, log)
+        assert isinstance(node.capability.gpu_count, int), (
```
Comment (LiliDeng) [inline lisa/microsoft/testsuites/gpu/gpusuite.py]: @kamalca 
We want to ensure that the maximum GPU count offered by Azure is correctly validated.
How can we achieve this in our situation?

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: If I understand correctly, it looks like the number of GPUs is a SKU quality, not a deployment parameter. So maybe GPU test cases are not a good candidate for this type of requirement.

I will have to ask Umang and Johnson what the motivation was for targeting the GPU test cases.
I will get back to you on this.

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: @LiliDeng I removed the changes to max_gpu test case

Comment (LiliDeng) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: the same case for the cases with name containing max

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: The goal of the other test requirements is different since they are deployment parameters. The goal is to test the max resource capabilities of a given SKU in many cases.

The old test requirements have a similar problem; for example, we hard coded 8 NICs but some SKUs now support 15 or more. This PR addresses this by testing the max resource for the SKU selected.

Can you clarify further your concerns?

Comment (LiliDeng) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: My concern is that we need to ensure the maximum count is properly validated.

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: Okay, so are you saying that you are not okay with lowering the minimum on the test case requirements?

Do I need to duplicate all of these test cases – one for the maximum on the given SKU, one for maximum allowed in Azure?

We need to be able to test the max_disk, max_nic capabilities of VM SKUs even if they are not capable of the max in Azure. In other words, we need to be able to test the max capabilities of smaller SKUs.

Can you help me out by describing exactly what part of my approach you have a problem with? What specific testing scenario are you worried about: chaos testing, image testing?
Do you have any ideas for how we can enable the scenario I am trying to achieve in a way that you are comfortable with?

Comment (LiliDeng) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: I am thinking whether we can use a switch to support this scenario.

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: I would like to avoid a switch. Every single validation flow would need to be addressed to understand whether they are testing the overall Azure Platform or if they are testing a specific SKU.
I worry that most people, especially people onboarding to LISA, do not know how test requirements work well enough to set a flag like this. There is no easy default behavior since both scenarios are common.

Comment (kamalca) [reply lisa/microsoft/testsuites/gpu/gpusuite.py]: Also, these test cases are not required to be running on Azure Platform – so for other platforms, isn't the requirement a bit arbitrary? For example, the minimum might be set too high for AWS or another platform.

I think it makes sense to either 
1. have duplicate testcases specific to Azure if we are going to tie the requirements to Azure specific capabilities
1. set the min nic / disk requirement in the runbook for testing where we want 8 NICs or 12 NVMe disks, etc.

**Comment by LiliDeng** [comment]
@kamalca I met regression when ran against this PR.

```
 Sriov.verify_sriov_max_vf_connection: FAILED   deployment failed. LisaException: VmSizeRestrictedToSingleNicWithAcceleratedNetworkingEnable: Virtual machine /subscriptions/0cc2a67a-58b9-4e4f-98a8-bfa46a28e896/resourceGroups/lisa-Azure-Adhoc-20251212-064035-049-e146/providers/Microsoft.Compute/virtualMachines/lisa-Azure-Adhoc-20251212-064035-049-e146-n1 with size Standard_D2ahs_v4 is restricted to have only one network interface with accelerated networking enabled.
VmSizeRestrictedToSingleNicWithAcceleratedNetworkingEnable: Virtual machine /subscriptions/0cc2a67a-58b9-4e4f-98a8-bfa46a28e896/resourceGroups/lisa-Azure-Adhoc-20251212-064035-049-e146/providers/Microsoft.Compute/virtualMachines/lisa-Azure-Adhoc-20251212-064035-049-e146-n0 with size Standard_D2ahs_v4 is restricted to have only one network interface with accelerated networking enabled.
```

**Review by LiliDeng — APPROVED** [review]
